File: core/tests/fixtures/mothers/contact_info_mother.py
from src.domain.customer import ContactInfo
from tests.fixtures.mothers.base import BaseMother


class ContactInfoMother(BaseMother):
    """Mother class for creating ContactInfo instances for testing purposes."""

    _DOE_CONTACT_INFO = {
        'email': 'johndoe@gmail.com',
        'phone1': '+1234567890',
        'phone2': '+0987654321',
    }

    _SMITH_CONTACT_INFO = {
        'email': 'jsmith@gmail.com',
        'phone1': '+1111111111',
        'phone2': '+2222222222',
    }

    @classmethod
    def doe(cls, **overrides) -> ContactInfo:
        """Returns a ContactInfo instance with the data of DOE_CONTACT_INFO."""
        return cls._create(cls._DOE_CONTACT_INFO, overrides)

    @classmethod
    def smith(cls, **overrides) -> ContactInfo:
        """Returns a ContactInfo instance with the data of SMITH_CONTACT_INFO."""
        return cls._create(cls._SMITH_CONTACT_INFO, overrides)

    @classmethod
    def _create(cls, values: dict, overrides: dict) -> ContactInfo:
        """Creates a ContactInfo instance with the given overrides."""
        data = cls._merge(values, overrides)
        return ContactInfo(**data)

File: editor/tests/fixtures/mothers/contact-info.mother.ts
import { BaseMother } from '@tests/fixtures/mothers/base/base-mother';
import { ContactInfo } from '@/customers/model/contact-info';

const DOE_CONTACT_INFO = {
  email: 'johndoe@gmail.com',
  phone1: '+1234567890',
  phone2: '+0987654321',
};

type ContactInfoFixture = typeof DOE_CONTACT_INFO;

export class ContactInfoMother extends BaseMother {
  static doe(this: void, overrides?: Partial<ContactInfo>): ContactInfo {
    return ContactInfoMother.create(DOE_CONTACT_INFO, overrides);
  }

  protected static create(
    values: ContactInfoFixture,
    overrides?: Partial<ContactInfo>,
  ): ContactInfo {
    const fields = ContactInfoMother.merge<ContactInfoFixture, ContactInfo>(
      values,
      overrides,
    );

    return ContactInfo.from(fields);
  }
}
